input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go

```diff
@@ -27,7 +27,9 @@ func Reslice[Slice ~[]model, model any](slice Slice, n int) Slice {
 	if n > cap(slice) {
 		slice = slices.Grow(slice, n-len(slice))
 	}
-	return slice[:n]
+	res := make(Slice, n)
+	copy(res, slice[:n])
+	return res
```

Comment:
This can create multiple allocations of slice (one in Grow and the other one which we are doing here). IIANM, the whole point of this function was to reduce allocation and GC (even when the capacity is higher) so I don't think this change is needed. OTOH, now that we are moving away from pooling, we might not even require Reslice but that would be a bigger change overall.

Reply:
Okay, I see. Thanks. I'll close this.